river/Output.zig

```diff
@@ -185,227 +163,259 @@ pub fn sendViewTags(self: Self) void {
+                    Log.Error.log("Invalid layout: Invalid window Y position.", .{});
+                    return false;
+                };
+            },
 
-        // Reduce size to allow space for borders/padding
-        const delta_size = (border_width + view_padding) * 2;
-        new_box.width -= delta_size;
-        new_box.height -= delta_size;
+            // Width
+            2 => {
+                box.width = std.fmt.parseInt(u32, token, 10) catch |err| {
+                    Log.Error.log("Invalid layout: Invalid window width.", .{});
+                    return false;
+                };
+            },
 
-        // Set the view's pending box to the new dimensions
-        view.pending_box = new_box;
+            // Height
+            3 => {
+                box.height = std.fmt.parseInt(u32, token, 10) catch |err| {
+                    Log.Error.log("Invalid layout: Invalid window height.", .{});
+                    return false;
+                };
+            },
 
+            else => {},
+        }
         i += 1;
     }
-}
-
-/// Wrapper for default layout with master area on the top
-pub fn layoutTopMaster(self: *Self, visible_count: u32, output_tags: u32) void {
-    layoutMasterStack(self, visible_count, output_tags, MasterPosition.Top);
-}
 
-/// Wrapper for default layout with master area on the right
-pub fn layoutRightMaster(self: *Self, visible_count: u32, output_tags: u32) void {
-    layoutMasterStack(self, visible_count, output_tags, MasterPosition.Right);
-}
+    if (i != 4) {
+        Log.Error.log("Invalid layout: Expected four window values, got {}.", .{i});
+        return false;
+    }
 
-/// Wrapper for default layout with master area on the bottom
-pub fn layoutBottomMaster(self: *Self, visible_count: u32, output_tags: u32) void {
-    layoutMasterStack(self, visible_count, output_tags, MasterPosition.Bottom);
+    Log.Debug.log("Parsed window layout: x={} y={} width={} height={}", .{ box.x, box.y, box.width, box.height });
+    return true;
 }
 
-/// Wrapper for default layout with master area on the left
-pub fn layoutLeftMaster(self: *Self, visible_count: u32, output_tags: u32) void {
-    layoutMasterStack(self, visible_count, output_tags, MasterPosition.Left);
+test "parse window configuration" {
+    const testing = @import("std").testing;
+    var box: Box = .{ .x = 0, .y = 0, .width = 0, .height = 0 };
+    testing.expect(parseWindowConfig("5 10 100 200", &box));
+    testing.expect(box.x == 5);
+    testing.expect(box.y == 10);
+    testing.expect(box.width == 100);
+    testing.expect(box.height == 200);
 }
 
-/// A layout in which every window uses the maximum available space.
-pub fn layoutFull(self: *Self, visible_count: u32, output_tags: u32) void {
+/// Execute an external layout function, parse its output and apply the layout
+/// to the output.
+fn layoutExternal(self: *Self, visible_count: u32, output_tags: u32) bool {
```

Comment:
Again, returning a `bool` to indicate failure isn't very idiomatic. Prefer a return type of `!void` to infer the error set, using `try functionThatMayFail()` in the body of `layoutExternal`, and a `catch`/`switch` handling the errors in the calling function.